### PR TITLE
6 create get endpoints

### PR DESCRIPTION
**Pull request by c0nsy**
## Describe your changes
1. Implemented GET endpoint for getting all records within the businesses collection in firebase
2. defined the Pydantic ORM model Business to handle interacting with the database
3. defined the Business schema to be used as a response model

## Issue ticket number and link

[Issue 6](https://github.com/Hermes-Marketing/hermes/issues/6)

## How to test changes
Navigate to the fast api docs to the business collection
trigger the endpoint with a valid collection name (businesses is the only one in the database)

Observe the response model, and confirm 200 status code.

## Checklist before requesting a review

- [x] I have performed a self-review of my code
- [x] If it is a core feature, I have added thorough tests.
- [ ] Do we need to implement analytics?

### CODE REVIEW

File: backend/app/api/v1/api.py
Comment (xbnrxout) [inline backend/app/api/v1/api.py:1]: usually with rename or typo fixes, thats its own PR as this usually results in some errors in prod test. So best to do these in a separate pr.

File: backend/app/api/v1/__pycache__/api.cpython-311.pyc
Comment (xbnrxout) [inline backend/app/api/v1/__pycache__/api.cpython-311.pyc:1]: With deletions its CRUCIAL to have a separate commit. If the changes are unrelated, they should each be in different commit

File: backend/app/api/v1/endpoints/business.py
Comment (xbnrxout) [inline backend/app/api/v1/endpoints/business.py:1]: you not only renamed, but added changes. Comparing to the deleted file is annoying

File: backend/app/core/main.py
Comment (xbnrxout) [inline backend/app/core/main.py:1]: Same commentary here. Hard to see the changes as isolated.

File: backend/app/core/main.py [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+            - Args: data: dict
+
+            - Returns:
+                The created record
+        """
+        object = None
+        try:
+            for key,value in data.items():
+                if(hasattr(self.model, key) and value is not None):
+                    setattr(object, key, value)
```
Comment (xbnrxout) [inline backend/app/core/main.py:33]: love the generic naming, well done

File: backend/app/core/main.py [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+            Generic method to create a single record
+            depending of the model assigned to the repository
+
+            - Args: data: dict
+
+            - Returns:
+                The created record
+        """
+        object = None
+        try:
```
Comment (xbnrxout) [inline backend/app/core/main.py:30]: You are doing a few things within create, make a separate function for verification, and for commiting, the failure for verifications gets rolled into any other failures interacting with the db

File: backend/app/models/business.py [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+    app.models.business
+    ~~~~~~~~~~~~~~~~~~~
+    Defines the Pydantic model for the Business resource
+"""
+
+from pydantic import BaseModel
+from typing import Optional
+
+
+class Business(BaseModel):
```
Comment (xbnrxout) [inline backend/app/models/business.py:11]: not sure on the naming here, while business is a representation of the "business"s meta, i believe we should name it something else. e.g. Enterprise, Company, Entity, etc....

File: backend/main.py [hunk 4]
```diff
@@ -2,17 +2,26 @@
...
     title="Hermes API",
     version="1.0.0",
     docs_url="/docs",
 )
 
-app.mount("/", StaticFiles(directory="../frontend/dist", html=True), name="static")
+# Use a relative path for the Docker container
+static_dir = "frontend/dist"
+if not os.path.exists(static_dir):
+    raise RuntimeError(f"Directory '{static_dir}' does not exist")
```
Comment (xbnrxout) [inline backend/main.py:16]: FileNotFoundError

File: backend/main.py [hunk 5]
```diff
@@ -2,17 +2,26 @@
...
+
+app.mount("/static", StaticFiles(directory=static_dir, html=True), name="static")
```
Comment (xbnrxout) [inline backend/main.py:18]: What files are we looking for?

**Review by xbnrxout — CHANGES_REQUESTED** [review]

File: backend/app/api/v1/api.py
Comment (c0nsy) [reply backend/app/api/v1/api.py:1]: Will do that going forward

File: backend/app/api/v1/__pycache__/api.cpython-311.pyc
Comment (c0nsy) [reply backend/app/api/v1/__pycache__/api.cpython-311.pyc:1]: Sounds good going forward ill apply that pattern

File: backend/app/models/business.py (hunk 3 above)
Comment (c0nsy) [reply backend/app/models/business.py:11]: Ill rename to company

File: backend/main.py (hunk 5 above)
Comment (c0nsy) [reply backend/main.py:18]: Honestly uvicorn was not executing without this block, maybe in a future ticket we can look more at this. What do you think?

File: backend/main.py (hunk 4 above)
Comment (c0nsy) [reply backend/main.py:16]: responded in another comment

**Review by xbnrxout — APPROVED** [review]

**Review by xbnrxout — APPROVED** [review]

**Review by xbnrxout — APPROVED** [review]
